includes/blocks/course-theme/class-course-progress-counter.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * File containing the Course_Progress class.
+ *
+ * @package sensei
+ * @since
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+
+/**
+ * Class Course_Progress is responsible for rendering the '< Prev Lesson | Next Lesson >' blocks.
+ */
+class Course_Progress_Counter {
+	/**
+	 * Course_Progress constructor.
+	 */
+	public function __construct() {
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-course-progress-counter',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render() : string {
+		$post   = get_post();
+		$course = $post;
+
+		if ( 'lesson' === $post->post_type ) {
+			$course_id = \Sensei()->lesson->get_course_id( $post->ID );
+			$course    = get_post( $course_id );
+		}
```

Comment:
I recommend `Sensei_Utils::get_current_course` here, which also works for quizzes. And an early return if the course is not found.

Reply:
Thanks for the tip! Updated. 👍